### ISSUE TITLE
Change dump(..., default_flow_style=...) default to False

### ISSUE THREAD

**Issue by ingydotnet** [yaml/pyyaml#199]
This is probably the most annoying default in PyYAML. It's the only PyYAML FAQ
entry: https://pyyaml.org/wiki/PyYAMLDocumentation#frequently-asked-questions

Changing the default to always prefer indented/block style would be good, but
is not back compatible. Since we will be breaking backwards compat big time
with the upcoming safe-load-by-default changes, this change should probably be
timed with that one.

**Comment by jason-s** [comment]
Hmm. It looks like there are actually *three* flow styles: True, False, and None:

```
>>> a = dict(a=dict(b=dict(a = ['a','b']), foo=45), c=dict(z=[1,2,3],yy=[4,5,6,7],zozo='Hey!'))
>>> print yaml.safe_dump(a)
a:
  b:
    a: [a, b]
  foo: 45
c:
  yy: [4, 5, 6, 7]
  z: [1, 2, 3]
  zozo: Hey!

>>> print yaml.safe_dump(a, default_flow_style=False)
a:
  b:
    a:
    - a
    - b
  foo: 45
c:
  yy:
  - 4
  - 5
  - 6
  - 7
  z:
  - 1
  - 2
  - 3
  zozo: Hey!

>>> print yaml.safe_dump(a, default_flow_style=True)
{a: {b: {a: [a, b]}, foo: 45}, c: {yy: [4, 5, 6, 7], z: [1, 2, 3], zozo: Hey!}}
```

**Comment by FichteFoll** [comment]
Fixed in https://github.com/yaml/pyyaml/commit/f20936573840842954b933801abeeaedad977b41, unless I am mistaken.

**Comment by perlpunk** [comment]
yep, fixed by #256

**Comment by Ark-kun** [comment]
None != False
I like the default_flow_style=None formatting  the best.
